src/pages/dashboard/index.tsx

```diff
@@ -1,15 +1,151 @@
+import List from "../../components/List";
+import dashboard from "../../api/dashboard";
 
 export const DashboardPage = () => {
-  const [data] = useState(null);
+  /**
+   * Set our initial useStates
+   */
+  const [items, setItems] = useState([]);
+  const [filteredItems, setFilteredItems] = useState([]);
+  const [availableItems, setAvailableItems] = useState([]);
+  const [myListItems, setMyListItems] = useState([]);
+  const [filterValue, setFilterValue] = useState("");
 
-  return data === null ? null : (
+  /**
+   * When data is done loading from the API use setState to assign
+   * initial state data for each of our Items Lists.
+   */
+  useEffect(() => {
+    const loadData = async () => {
+      let deviceItems = await dashboard().getData().then(result => {
+        return result;
+      });
+      setItems(deviceItems);
+      setFilteredItems([...deviceItems]);
+      setAvailableItems([...deviceItems]);
+    }
+    loadData();
+  }, []);
+
+  /**
+   * When availableItems, myListItems or filterValue changes run
+   * filterAvailableDevices and pass in filterValue
+   */
+  useEffect(() => {
+    filterAvailableDevices(filterValue);
+  }, [availableItems, myListItems, filterValue]);
+
+  /**
+   * Add Device to My List Items and Remove from Available List
+   * @param id
+   */
+  const addDevice = (id) => {
+    const temp = [...availableItems];
+    setAvailableItems(temp.filter((item) => item.id !== id));
+    const myDeviceItem = temp.filter((item) => item.id === id);
+    setMyListItems([...myListItems, ...myDeviceItem]);
+  }
+
+  /**
+   * Remove Device From My List and add the the Available List
+   * @param id
+   */
+  const removeDevice = (id) => {
+    const temp = [...myListItems];
+    const myDeviceItem = temp.filter((item) => item.id === id);
+    setAvailableItems([...availableItems, ...myDeviceItem]);
+    setMyListItems(temp.filter((item) => item.id !== id));
+  }
+
+  /**
+   * On Keyup of the filter input take the value and
+   * setFilterValue with the value of the input
+   * @param event
+   */
+  const filterChanged = (event) => {
+    let value = event.target.value;
+    //filterAvailableDevices(value);
+    setFilterValue(value);
+  }
+
+  /**
+   * If we have a filter value, were looping through all
+   * the keys in each availableItem and checking if the value of the key
+   * includes some words or text of the filter input value. If it does, we push it to
+   * a temporary array (foundItems) and then use that to assign setFilteredItems.
+   * If there is no filter value it just sets filteredItems to a clone of available items
+   * @param value
+   */
+  const filterAvailableDevices = (value) => {
+    let foundItems = [];
+
+    if (!value) {
+      setFilteredItems([...availableItems]);
+    } else {
+      availableItems.forEach((item) => {
+        let itemFound = false;
+        for (let key in item) {
+          let itemValue = String(item[key]);
```

Comment:
This technically works, but I would do `item[key]`.toString()` rather than just cast it as a String.